Sources/Turf/GeoJSON.swift

```diff
@@ -89,3 +89,36 @@ extension Feature: GeoJSONObjectConvertible {
 extension FeatureCollection: GeoJSONObjectConvertible {
     public var geoJSONObject: GeoJSONObject { return .featureCollection(self) }
 }
+
+/**
+ A GeoJSON object that can contain [foreign members](https://datatracker.ietf.org/doc/html/rfc7946#section-6.1) in arbitrary keys.
+ */
+public protocol ForeignMemberContainer {
```

Comment:
I think client code should have access to the foreign members through the `foreignMembers` property, just as they would when using JSONSerialization instead of this library. It would also make it possible for the developer to build and write out GeoJSON source code that contains foreign members without having to go through JSONSerialization.

Reply:
That makes sense. I was more thinking about whether the properties on the implementers could be public while keeping the protocol internal.

Reply:
If we could do that, we’d still have to duplicate the documentation for `foreignMembers` across all the types that implement it. In general, I think it would be kind of nice to move the various aspects of these GeoJSON types into protocols, so that developers could write more generic spatial methods targeting these types.

Reply:
Yep, that's fine with me